api/src/modules/h3-data/h3-data.repository.ts

```diff
@@ -438,46 +435,59 @@ export class H3DataRepository extends Repository<H3Data> {
     originIds?: string[],
     supplierIds?: string[],
   ): Promise<{ riskMap: H3IndexValueData[]; quantiles: number[] }> {
-    const query: SelectQueryBuilder<any> = getManager()
+    const queryBuilder: SelectQueryBuilder<any> = getManager()
       .createQueryBuilder()
-      .select(
-        `h3_to_parent(unnest(gr."h3Flat"::h3index[]), ${resolution})`,
-        'h',
-      )
-      .addSelect('sum(ir.value/gr."h3FlatLength")', 'v')
+      .select(`h3_to_parent((h3data.h3index), ${resolution})`, `h`)
+      .addSelect(`sum(h3data.value)`, `v`)
       .from(SourcingLocation, 'sl')
       .innerJoin(SourcingRecord, 'sr', 'sl.id = sr.sourcingLocationId')
       .innerJoin(IndicatorRecord, 'ir', 'sr.id = ir.sourcingRecordId')
-      .innerJoin(GeoRegion, 'gr', 'sl.geoRegionId = gr.id')
-      .where('sl.scenarioInterventionId IS NULL')
-      .andWhere('sl.interventionType IS NULL')
-      .andWhere('ir.indicatorId = :indicatorId', { indicatorId: indicator.id })
-      .andWhere('sr.year = :year', { year })
-      .andWhere('gr."h3FlatLength" > 0')
-      .groupBy('h');
+      .innerJoin('material_to_h3', 'mth', 'mth.materialId = sl.materialId');
+
+    const query: string = queryBuilder.getSql();
+
+    const sqlQueryParams: any[] = [year, indicator.id];
+
+    let sqlQuery: string =
+      query +
+      `, LATERAL(SELECT h3index, ir.value/ir.scaler * value as value FROM get_h3_data_over_georegion(sl."geoRegionId", mth."h3DataId")) h3data
+      WHERE sr.year=$1 
+      AND sl."scenarioInterventionId" IS NULL
+      AND sl."interventionType" IS NULL
+      AND ir."indicatorId"=$2`;
+
+    let numberOfParams: number = 2;
 
     if (materialIds) {
-      query.andWhere('sl.material IN (:...materialIds)', { materialIds });
+      sqlQuery =
+        sqlQuery + ` AND sl."materialId" = ANY ($${numberOfParams + 1})`;
```

Comment:
wouldn't be easier to build up conditionally the query using the queryBuilder instead of manipulating a string? Feels a bit risky